.github/workflows/snap-store-publish-to-edge.yml

```diff
@@ -0,0 +1,251 @@
+    outputs:
+      issue_number: ${{ steps.comment.outputs.number }}
+
+  # Deploy the snap inside a desktop VM and grab screenshots of the desktop, and of
+  # the application, then post them as a comment on the issue created above.
+  grab-screenshots:
+    name: Gather screenshots
+    environment: "Edge Branch"
+    needs: 
+      - build
+      - create_issue
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v4
+
+      - name: Setup LXD
+        uses: canonical/setup-lxd@v0.1.1
+
+      - name: Prepare VM
+        run: |
+          .github/vmctl prepare
+          .github/vmctl exec "sudo snap install ${{ env.SNAP_NAME }} --channel latest/edge"
+          .github/vmctl exec "snap run ${{ env.SNAP_NAME }} &>/home/ubuntu/${{ env.SNAP_NAME }}.log &"
+          sleep 20
+
+      - name: Gather screenshots
+        run: |
+          .github/vmctl screenshot-full
+          .github/vmctl screenshot-window
+
+      - name: Output application logs
+        run: |
+          .github/vmctl exec "cat /home/ubuntu/${{ env.SNAP_NAME }}.log"
+
+      - name: Checkout snapcrafters/ci-screenshots
+        uses: actions/checkout@v4
+        with: 
+          repository: snapcrafters/ci-screenshots
+          path: ci-screenshots
+          token: ${{ secrets.SNAPCRAFTERS_BOT_COMMIT }}
+
+      - name: Upload screenshots to snapcrafters/ci-screenshots
+        id: screenshots
+        run: |
+          file_prefix="$(date +%Y%m%d)-${{ env.SNAP_NAME }}-${{ needs.create_issue.outputs.issue_number }}"
+
+          pushd ci-screenshots
+          mv ../screenshot-screen.png "${file_prefix}-screen.png"
+          mv ../screenshot-window.png "${file_prefix}-window.png"
+
+          git config --global user.email "merlijn.sebrechts+snapcrafters-bot@gmail.com"
+          git config --global user.name "Snapcrafters Bot"
+
+          git add -A .
+          git commit -m "data: screenshots for snapcrafters/${{ env.SNAP_NAME }}#${{ needs.create_issue.outputs.issue_number }}"
+          git push origin main
+
+          echo "screen=https://raw.githubusercontent.com/snapcrafters/ci-screenshots/main/${file_prefix}-screen.png" >> "$GITHUB_OUTPUT"
+          echo "window=https://raw.githubusercontent.com/snapcrafters/ci-screenshots/main/${file_prefix}-window.png" >> "$GITHUB_OUTPUT"
+
+      - name: Comment on call for testing issue with screenshots
+        uses: actions/github-script@v7
+        with:
+          script: |
+            github.rest.issues.createComment({
+              issue_number: ${{ needs.create_issue.outputs.issue_number }},
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: `The following screenshots were taken during automated testing:
+
+              ![window](${{ steps.screenshots.outputs.window }})
+
+              ![screen](${{ steps.screenshots.outputs.screen }})
+              `
+            })
```

Comment:
Do we need this?

Reply:
No, i've not tested any of the GH action logic.  There's likely quite a bit of fluff left to remove.  Hence the Draft PR